src/main/java/org/folio/circulation/domain/policy/library/KeepCurrentDateStrategy.java

```diff
@@ -18,6 +18,11 @@ public KeepCurrentDateStrategy(DateTimeZone zone) {
 
   @Override
   public Result<DateTime> calculateDueDate(DateTime requestedDate, AdjacentOpeningDays openingDays) {
-    return succeeded(requestedDate.withZone(zone).withTime(END_OF_A_DAY));
+    final DateTime dueDate = requestedDate.withZone(zone)
+      // Always keep the requested date
+      .withDate(requestedDate.toLocalDate())
+      .withTime(END_OF_A_DAY);
+
+    return succeeded(dueDate);
```

Comment:
I think we can use `withZoneRetainFields` to achieve the same thing

```suggestion
  return succeeded(requestedDate.withZoneRetainFields(zone).withTime(END_OF_A_DAY));
```

Reply:
Thanks, I haven't seen this method before :)

Reply:
I hadn't until recently, I rather stumbled upon it as part of my work to move from Joda time to Java time